<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1131</width>
    <height>863</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="Canvas" name="main_openGL_widget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>550</width>
      <height>550</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="yawplus">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>640</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pitchplus">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>680</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rollplus">
    <property name="geometry">
     <rect>
      <x>100</x>
      <y>720</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>+</string>
    </property>
   </widget>
   <widget class="QPushButton" name="yawminus">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>640</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pitchminus">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>680</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rollminus">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>720</y>
      <width>30</width>
      <height>30</height>
     </rect>
    </property>
    <property name="text">
     <string>-</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>650</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Yaw</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>690</y>
      <width>41</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Pitch</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>720</y>
      <width>31</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Roll</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="subdivisionCountBox">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>680</y>
      <width>81</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>690</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Subdivision count:</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="iterationCountBox">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>720</y>
      <width>81</width>
      <height>31</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>730</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Iteration count:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="executeButton">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>730</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Execute</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="highlightEdgesCheckBox">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>590</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>Highlight edges</string>
    </property>
   </widget>
   <widget class="QComboBox" name="modelSelectionComboBox">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>640</y>
      <width>101</width>
      <height>51</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>630</y>
      <width>121</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>log-aesthetic alpha:</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="logAlphaSpinBox">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>630</y>
      <width>81</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="resetModelButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>740</y>
      <width>91</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Reset model</string>
    </property>
   </widget>
   <widget class="QPushButton" name="subdivideButton">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>680</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Subdivide</string>
    </property>
   </widget>
   <widget class="QComboBox" name="algSelectorComboBox">
    <property name="geometry">
     <rect>
      <x>120</x>
      <y>610</y>
      <width>103</width>
      <height>32</height>
     </rect>
    </property>
   </widget>
   <widget class="QOpenGLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>10</y>
      <width>550</width>
      <height>550</height>
     </rect>
    </property>
   </widget>
   <widget class="QProgressBar" name="algProgressBar">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>580</y>
      <width>271</width>
      <height>16</height>
     </rect>
    </property>
    <property name="value">
     <number>24</number>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>560</y>
      <width>99</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="loadModelPushButton">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>690</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Load model</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="vertexIdsCheckBox">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>590</y>
      <width>85</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex ids</string>
    </property>
   </widget>
   <widget class="QComboBox" name="curvatureSelectorComboBox">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>610</y>
      <width>103</width>
      <height>32</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="exportModelPushButton">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>720</y>
      <width>100</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Export model</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1131</width>
     <height>37</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Canvas</class>
   <extends>QOpenGLWidget</extends>
   <header>canvas.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
